src/main/java/net/openhft/chronicle/core/threads/EventHandler.java

```diff
@@ -34,28 +35,31 @@ default void eventLoop(EventLoop eventLoop) {
     }
 
     /**
-     * This handler has been added to an EventLoop or when the EventLoop starts.
-     * This is always called in the EventLoop thread.
-     * This is called after the first called to eventLoop() to make it clearer.
+     * This handler has begun being executed on the event loop. This will be called after the call to {@link #eventLoop(EventLoop)}
+     * and will always be called on the EventLoop thread.
+     * <p>
+     * This call will be followed by zero or more invocations of {@link #action()} and then an invocation of {@link #loopFinished()}
+     * <p>
+     * Exceptions thrown by loopStarted will be logged and the handler will be removed from the event loop.
      */
     default void loopStarted() {
     }
 
     /**
-     * Notify handler that the event handler's action method
-     * will not be called again. This is an appropriate place to perform cleanup.
-     * Event loop implementations call this once only, from the event loop's execution thread.
-     * <p>This is called either when the event loop is terminating, or if this EventHandler is being
+     * Notify handler that the event handler's action method will not be called again. It will be called
+     * if and only if {@link #loopStarted()} was called, and will always be called on the event loop thread.
+     * <p>
+     * This is called either when the event loop is terminating, or if this EventHandler is being
      * removed from the event loop.
-     * <p>If this implements {@link Closeable} then the event loop will call close (once only) on this after
+     * <p>
+     * If this handler implements {@link Closeable} then the event loop will call close (once only) on this after
      * loopFinished has been called.
-     * <p>If this implements {@link Closeable} and something other than the event loop
-     * calls close then it is expected that this will throw {@link InvalidEventHandlerException} next time
-     * {@link #action()} is called. This will then allow the event loop to call loopFinished (and close) on this.
-     * If this use case is required it is strongly recommend that this event handler guards against close
-     * being called more than once (and ignores subsequent calls).
-     * <p>Exceptions thrown by loopFinished or close are caught and logged (at debug level)
-     * and cleanup continues.
```

Comment:
This seems like an implementation detail and not a part of the contract we need to explain. If someone closes a handler, it's likely it'll throw some kind of exception when action is called, and we'll remove the handler. That much seems implicit in the contract.